### DISCUSSION TITLE
Mastodon shows only logo (Tried some fixes from other posts)

### DISCUSSION THREAD

**Discussion by akama-aka** [mastodon/mastodon#23036] [Q&A]
Hello, everytime I try to open mastodon I get this:
![image](https://user-images.githubusercontent.com/105487011/211558651-b6805412-9916-4b07-8d6f-7af6c60a82e6.png)
I already viewd and tried these solutions:
#22758,
#20747,
#21169

I hope someone can help me. If more informations needed just send what I should provide

**Comment by StrangeMinds1960** [comment]
Don't know if this will work for you (if you haven't already found it).  I'm sure this is the solution which fixed it for me

'I solved it by including www-data into the mastodon group and viceversa'

**Reply by akama-aka** [reply]
I linked another discussion in this message and I also tried this but its still not working
![image](https://user-images.githubusercontent.com/105487011/211566406-3d42d0e2-a636-4fe0-90bb-070f945afbba.png)

We also restarted the Server but still no changes

**Comment by rbairwell** [comment]
How have you installed Mastodon? From [source](https://docs.joinmastodon.org/admin/install/), from Docker, from DigitalOcean 1-click or some other way?

It looks like the CSS and Javascript hasn't been compiled - have you Installed the [latest dependencies](https://docs.joinmastodon.org/admin/install/#installing-the-last-dependencies) and then [generated the configuration](https://docs.joinmastodon.org/admin/install/#generating-a-configuration)? Are/were there any errors when you did so?

**Reply by akama-aka** [reply]
I downloaded from source and it should installed and compiled everything. But I can try again

**Reply by akama-aka** [reply]
The error at the bottom is because the user already exists in the database
![image](https://user-images.githubusercontent.com/105487011/212491158-48b28446-dd32-4196-8642-35e6ae0e865d.png)

But still same issue:

![image](https://user-images.githubusercontent.com/105487011/212491345-16ee7abc-3ceb-4ea7-84b6-1a8cce1249c6.png)

**Comment by selflessmind** [comment]
I had the same issue, following the steps from joinmastodon.org. After a lot of searching online it turned out the permissions needed tweaking to access the core files.  This is what I had to enter 
`chown mastodon:www-data /home/mastodon/`
`chmod 0710 -v /home/mastodon/`

**Reply by akama-aka** [reply]
This also doesnt work :/

**Reply by flth** [reply]
I'm in the same place, was there any solution?

**Reply by akama-aka** [reply]
nope

**Reply by flth** [reply]
`chmod +x /home/mastodon` should help.
Fixed it for me right now

**Reply by akama-aka** [reply]
Nope

**Reply by snullp** [reply]
This is because nginx unable to read /home/mastodon directory. chmod +rx /home/mastodon/ works for me.

**Reply by leopo17** [reply]
Using this works for me. Server DigirtalOcean Mastodon Image Droplet. Debian Mastodon 4.2.1, Debian 6.1.
Thanks for help

**Comment by shehackedyou** [comment]
I have the same issue, works fine in development mode. So I assume it has something to do with assets not being prepared correctly.

**Reply by shehackedyou** [reply]
`RAILS_SERVE_STATIC_FILES=true RAILS_ENV=production bundle exec rails server `

This solved it for me, you can add the `RAILS_SERVER_STATIC_FILES` env to your production.env file, or alias it, or ~/.bashrc since it is outside the project. 

There are a lot of solutions each with its own caveats. 

Probably best to stick with the production.env file; I would presume it is already in the .gitignore. 

I'm pretty confident that will help you too; these wack ideas of changing the permissions on a mastodon user that shouldn't even have a user folder is a bit mind boggling.

**Reply by snullp** [reply]
Can you try chmod +rx /home/mastodon/ while keep using nginx. This is because nginx unable to read /home/mastodon directory.
